### PR TITLE
Add throttling for image install updates

### PR DESCRIPTION
**Pull request by Page-**
Change-type: patch

### CODE REVIEW

**Comment by Page-** [comment]
@balena-ci I self-certify!

File: src/lib/config.ts [hunk 1]
```diff
@@ -243,6 +243,9 @@ export const HIDE_UNVERSIONED_ENDPOINT = boolVar(
 export const METRICS_MAX_REPORT_INTERVAL_SECONDS = Math.round(
 	intVar('METRICS_MAX_REPORT_INTERVAL', 1 * MINUTES) / 1000,
 );
+export const DOWNLOAD_PROGRESS_MAX_REPORT_INTERVAL_SECONDS = Math.round(
+	intVar('DOWNLOAD_PROGRESS_MAX_REPORT_INTERVAL_SECONDS', 30) / 1000,
```
Comment (thgreasi) [inline src/lib/config.ts:247]: So the default value is 30ms?
Should this be:
```
intVar('DOWNLOAD_PROGRESS_MAX_REPORT_INTERVAL', 30) / 1000,
```
or
```
intVar('DOWNLOAD_PROGRESS_MAX_REPORT_INTERVAL_SECONDS', 30 * SECONDS) / 1000,
```

File: src/features/device-state/routes/state-patch.ts [hunk 2]
```diff
@@ -256,6 +261,57 @@ const shouldMetricsUpdate = (() => {
...
+			lastReport.status !== body.status ||
+			// Or if the release has changed
+			lastReport.is_provided_by__release !== body.is_provided_by__release ||
+			// Or if the download progress has hit a milestone...
+			// From not downloading to downloading
+			(lastReport.download_progress == null &&
+				body.download_progress != null) ||
+			// From downloading to not downloading
+			(lastReport.download_progress != null &&
+				body.download_progress == null) ||
```
Comment (thgreasi) [inline src/features/device-state/routes/state-patch.ts:299]: Had to read this twice to get it.
We can combine those two to v right?
```
(lastReport.download_progress == null) != (body.download_progress == null)
```

File: src/features/device-state/routes/state-patch.ts [hunk 3]
```diff
@@ -256,6 +261,57 @@ const shouldMetricsUpdate = (() => {
...
+			// Hits 100%
+			body.download_progress === 100
+		) {
+			// And we add a new entry
+			await lastImageInstallReport.set(key, {
+				download_progress: lastReport?.download_progress,
```
Comment (thgreasi) [inline src/features/device-state/routes/state-patch.ts:305]: Why are we only assigning the `download_progress`?
Could/Shouldn't this be just destructuring the whole `lastReport`?

File: src/lib/config.ts (hunk 1 above)
Comment (thgreasi) [reply src/lib/config.ts:247]: Oh I see that https://github.com/balena-io/open-balena-api/pull/887 fixed that :+1:

File: src/features/device-state/routes/state-patch.ts (hunk 3 above)
Comment (Page-) [reply src/features/device-state/routes/state-patch.ts:305]: Right I meant to add a comment to this as well, but basically we want to keep the last reported download progress if the current report doesn't include it

File: src/features/device-state/routes/state-patch.ts (hunk 2 above)
Comment (Page-) [reply src/features/device-state/routes/state-patch.ts:299]: We can combine them but I had them separate to follow the comment chain cleanly

File: src/features/device-state/routes/state-patch.ts (hunk 3 above)
Comment (Page-) [reply src/features/device-state/routes/state-patch.ts:305]: https://github.com/balena-io/open-balena-api/pull/888

Comment (thgreasi) [reply src/features/device-state/routes/state-patch.ts:305]: @Page- my question was mostly why we care only about the `download_progress` and we didn't do:
```
				...lastReport,
				...body,
				updateTime: now,
```

**Pull request merged**